.github/workflows/main.yml

```diff
@@ -1,32 +1,107 @@
 name: API Manager Build with Test
 
-on: [push, pull_request]
+on:
+  pull_request:
+    branches:
+      - master
+  workflow_dispatch:
+    branches:
+      - master
```

Comment:
Same here.

Bdw we can add a new github workflow to build PRs and run with tests.

Reply:
https://github.com/wso2/carbon-apimgt/pull/10126/commits/f31a26dbfdd8436f0c2a6d20b334441d2772a183